No bin/prepare_refseqs.pl found

The instructions say: 

> The data generation pipeline is based on the typical jbrowse commands such as prepare-refseqs.pl and flatfile-to-json.pl, and these scripts are automatically copied to a local bin/ directory when you run the setup scripts (e.g. apollo run-local or apollo deploy or install_jbrowse.sh).
> 
> If you don't see a bin/ subdirectory containing these scripts after running the setup, check setup.log and check the troubleshooting guide for additional tips or feel free to post the error and setup.log on GitHub or the mailing list._

But after running apollo run-local there is no bin directory created. No setup.log file was created. The troubleshooting guide does not cover this topic. So I am posting this problem to GitHub as suggested in the guide. Thank you!

```
ls web-app/:
annotator  css  images  jbrowse  js  META-INF  translation_tables  WEB-INF

ls web-app/jbrowse/:
plugins

ls jbrowse-download/:
plugins
```

ant clean-all then ant run-local:

```
Buildfile: build.xml

BUILD FAILED
Target "run-local" does not exist in the project "Apollo". 

Total time: 0 seconds
```

Here is the end of the output of apollo run-local:

```
BUILD FAILED
/usr/local/src/Apollo-2.0.2/build.xml:235: Failed to copy /usr/local/src/Apollo-2.0.2/jbrowse-download/tests/data/SL2.40_all_rna_seq.v1.bigwig to /usr/local/src/Apollo-2.0.2/web-app/jbrowse/tests/data/SL2.40_all_rna_seq.v1.bigwig due to java.io.FileNotFoundException /usr/local/src/Apollo-2.0.2/jbrowse-download/tests/data/SL2.40_all_rna_seq.v1.bigwig (No such file or directory)
```

I installed the new Git system-wide, replacing the previous version completely.

```
git --version
git version 2.8.0.GIT
```


```
Buildfile: build.xml

copy.apollo.plugin.webapp:

setup-jbrowse:

git.clone:
     [exec] Submodule 'src/FileSaver' (git://github.com/dkasenberg/FileSaver.js.git) registered for path 'src/FileSaver'
     [exec] Submodule 'src/dbind' (git://github.com/rbuels/dbind.git) registered for path 'src/dbind'
     [exec] Submodule 'src/dgrid' (git://github.com/SitePen/dgrid.git) registered for path 'src/dgrid'
     [exec] Submodule 'src/dijit' (https://github.com/dojo/dijit.git) registered for path 'src/dijit'
     [exec] Submodule 'src/dojo' (https://github.com/dojo/dojo.git) registered for path 'src/dojo'
     [exec] Submodule 'src/dojox' (https://github.com/dojo/dojox.git) registered for path 'src/dojox'
     [exec] Submodule 'src/jDataView' (git://github.com/rbuels/jDataView.git) registered for path 'src/jDataView'
     [exec] Submodule 'src/json-schema' (git://github.com/kriszyp/json-schema.git) registered for path 'src/json-schema'
     [exec] Submodule 'src/jszlib' (git://github.com/rbuels/jszlib.git) registered for path 'src/jszlib'
     [exec] Submodule 'src/lazyload' (git://github.com/rbuels/lazyload.git) registered for path 'src/lazyload'
     [exec] Submodule 'src/put-selector' (git://github.com/kriszyp/put-selector.git) registered for path 'src/put-selector'
     [exec] Submodule 'src/util' (https://github.com/dojo/util.git) registered for path 'src/util'
     [exec] Submodule 'src/xstyle' (git://github.com/kriszyp/xstyle.git) registered for path 'src/xstyle'
     [exec] Result: 1

copy.apollo.plugin.jbrowse:
     [echo] copying plugin to jbrowse git ...
     [copy] Copying 190 files to /usr/local/src/Apollo-2.0.2/jbrowse-download/plugins/WebApollo
     [echo] Will be setting up jbrowse
     [copy] Copying 1357 files to /usr/local/src/Apollo-2.0.2/web-app/jbrowse

BUILD FAILED
/usr/local/src/Apollo-2.0.2/build.xml:235: Failed to copy /usr/local/src/Apollo-2.0.2/jbrowse-download/tests/data/SL2.40_all_rna_seq.v1.bigwig to /usr/local/src/Apollo-2.0.2/web-app/jbrowse/tests/data/SL2.40_all_rna_seq.v1.bigwig due to java.io.FileNotFoundException /usr/local/src/Apollo-2.0.2/jbrowse-download/tests/data/SL2.40_all_rna_seq.v1.bigwig (No such file or directory)

Total time: 1 minute 10 seconds


ls -l jbrowse-download/tests/data/SL2.40*
lrwxrwxrwx 1 root root     51 May  5 14:31 jbrowse-download/tests/data/SL2.40_all_rna_seq.v1.bigwig -> /home/rob/jbrowse/data/SL2.40_all_rna_seq.v1.bigwig
-rw-r--r-- 1 root root 426336 May  5 14:31 jbrowse-download/tests/data/SL2.40ch10_sample.gff3
```

I upgraded ant to V1.9.7, but test (1) didn't work with the same result as before

So I tried (2) and clone the master branch. Reading the git documents seemed like this was the way:

```
git clone https://github.com/GMOD/Apollo.git
cd Apollo
git checkout master^^
```

Then ./apollo run-local gave this result which looks very different than before:
```

which: no gradle in (/root/.sdkman/candidates/grails/current/bin:/usr/local/google-cloud-sdk/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin)
Gradle not found using gradlew
Defining custom 'build' task when using the standard Gradle lifecycle plugins has been deprecated and is scheduled to be removed in Gradle 3.0
Defining custom 'clean' task when using the standard Gradle lifecycle plugins has been deprecated and is scheduled to be removed in Gradle 3.0
final JBrowse settings [git:[url:https://github.com/GMOD/jbrowse, branch:master, alwaysPull:false, alwaysRecheck:false], plugins:[WebApollo:[included:true], NeatHTMLFeatures:[included:true, linearGradient:0], NeatCanvasFeatures:[included:true], RegexSequenceSearch:[included:true], HideTrackLabels:[included:true]]]
final plugins [WebApollo:[included:true], NeatHTMLFeatures:[included:true, linearGradient:0], NeatCanvasFeatures:[included:true], RegexSequenceSearch:[included:true], HideTrackLabels:[included:true]]
Copying apollo plugin
handling jbrowse release [git:[url:https://github.com/GMOD/jbrowse, branch:master, alwaysPull:false, alwaysRecheck:false], plugins:[WebApollo:[included:true], NeatHTMLFeatures:[included:true, linearGradient:0], NeatCanvasFeatures:[included:true], RegexSequenceSearch:[included:true], HideTrackLabels:[included:true]]]
handling jbrowse [git:[url:https://github.com/GMOD/jbrowse, branch:master, alwaysPull:false, alwaysRecheck:false], plugins:[WebApollo:[included:true], NeatHTMLFeatures:[included:true, linearGradient:0], NeatCanvasFeatures:[included:true], RegexSequenceSearch:[included:true], HideTrackLabels:[included:true]]]
:evaluateJBrowseConfigs UP-TO-DATE
:installJBrowse
installing jbrowse [git:[url:https://github.com/GMOD/jbrowse, branch:master, alwaysPull:false, alwaysRecheck:false], plugins:[WebApollo:[included:true], NeatHTMLFeatures:[included:true, linearGradient:0], NeatCanvasFeatures:[included:true], RegexSequenceSearch:[included:true], HideTrackLabels:[included:true]]]
Cloning: https://github.com/GMOD/jbrowse into jbrowse-download.
using bower to install: GMOD/jbrowse#master
:installJBrowse FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':installJBrowse'.
> A problem occurred starting process 'command 'bower''

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 5.281 secs
```

--- a/docs/Troubleshooting.md
+++ b/docs/Troubleshooting.md
@@ -82,11 +82,11 @@ http://tomcat.apache.org/tomcat-7.0-doc/security-howto.html#Non-Tomcat_settings
 
 #### JBrowse tools don't show up in ```bin``` directory
 
-If the ```bin``` directory with jbrowse tools doesn't show up after calling ```install_jbrowse.sh``` and during the build phase you see something like this:
+If the ```bin``` directory with JBrowse tools doesn't show up after calling ```install_jbrowse.sh``` JBrowse is having trouble installing itself for a few possible reasons.   If these do not work, please observe the [troubleshooting](http://jbrowse.org/code/JBrowse-1.12.1/docs/tutorial/#Troubleshooting) version of JBrowse a well.   
 
 ##### Git tool is too old
 
-Git expects to clone a single branch which is supported in git 1.7.10 and greater.  The output when that fails looks like this.
+Git expects to clone a single branch which is supported in git 1.7.10 and greater.  The output when that fails looks something like this:
 
 ```
 Buildfile: build.xml
@@ -99,7 +99,7 @@ git.clone:
 [exec] Result: 129
 ```
 
-The solution is to upgrade git to 1.7.10 or greater or remove the ```--single-branch``` option in build.xml. 
+The solution is to upgrade git to 1.7.10 or greater or remove the line with the ```--single-branch``` option in ```build.xml```. 
 
 ##### Accessing git behind a firewall. 
 

--- a/docs/Troubleshooting.md
+++ b/docs/Troubleshooting.md
@@ -80,6 +80,52 @@ http://tomcat.apache.org/tomcat-7.0-doc/security-howto.html#Non-Tomcat_settings
 
 ### Errors with JBrowse
 
+#### JBrowse tools don't show up in ```bin``` directory
+
+If the ```bin``` directory with jbrowse tools doesn't show up after calling ```install_jbrowse.sh``` and during the build phase you see something like this:
+
+##### Git tool is too old
+
+Git expects to clone a single branch which is supported in git 1.7.10 and greater.  The output when that fails looks like this.
+
+```
+Buildfile: build.xml
+
+copy.apollo.plugin.webapp:
+
+setup-jbrowse:
+
+git.clone:
+[exec] Result: 129
+```
+
+The solution is to upgrade git to 1.7.10 or greater or remove the ```--single-branch``` option in build.xml. 
+
+##### Accessing git behind a firewall. 
+
+If you are behind a firewall, checking out code using the ```git://``` protocol may not be allowed, but that is the default.    The output will look something like this:
+
+```
+setup-jbrowse:
+
+git.clone:
+     [exec] Submodule 'src/FileSaver' (git://github.com/dkasenberg/FileSaver.js.git) registered for path 'src/FileSaver'
+     [exec] Submodule 'src/dbind' (git://github.com/rbuels/dbind.git) registered for path 'src/dbind'
+    . . . .
+     [exec] Submodule 'src/xstyle' (git://github.com/kriszyp/xstyle.git) registered for path 'src/xstyle'
+     [exec] Result: 1
+```
+
+with possibly more output below. 
+
+Type:
+
+```git config --global url."https://".insteadOf git://``` 
+
+in the command-line and then re-install using ```./apollo clean-all``` ```./apollo run-local``` (or deploy or release).
+
+
+
 #### e.g. "Can't locate Hash/Merge.pm in @INC" or "Can't locate JBlibs.pm in @INC"
 
 If you are trying to run the jbrowse binaries but get these sorts of errors, try running `install_jbrowse.sh` which will

